pyqtgraph/functions.py

```diff
@@ -1752,51 +1736,114 @@ def arrayToQPath(x, y, connect='all'):
             if first < len(x):
                 # Replace all non-finite entries from beginning of arr with the first finite one
                 idx[:first] = first
-                arr[1:-1] = arr[1:-1][idx]
+                arr[:] = arr[:][idx]
 
     # decide which points are connected by lines
     if eq(connect, 'all'):
-        arr[1:-1]['c'] = 1
+        polygon = arrayToQPolygonF(
+            arr['x'],
+            arr['y']
+        )
+        path.addPolygon(polygon)
+        return path
     elif eq(connect, 'pairs'):
-        arr[1:-1]['c'][::2] = 0
-        arr[1:-1]['c'][1::2] = 1  # connect every 2nd point to every 1st one
+        arr[:]['c'][::2] = 0
+        arr[:]['c'][1::2] = 1  # connect every 2nd point to every 1st one
     elif eq(connect, 'finite'):
         # Let's call a point with either x or y being nan is an invalid point.
         # A point will anyway not connect to an invalid point regardless of the
         # 'c' value of the invalid point. Therefore, we should set 'c' to 0 for
         # the next point of an invalid point.
-        if isfinite is None:
+        if qt6:
+            # 1) look for the first finite element.
             isfinite = np.isfinite(x) & np.isfinite(y)
-        arr[2:]['c'] = isfinite
+            first = np.argwhere(isfinite == np.True_)[0]
```

Comment:
A few issues here:
1) ```isfinite``` is already boolean, so no need to test against np.True_ once more
2) result of ```argwhere``` could be empty for the degenerate case of all non-finite. (Hence the "if found")
3) the caller doesn't expect the input arguments ```x``` and ```y``` to be modified

Reply:
ahh, yes, all good comments; should be better now.

I wish there was an easier way to break out the isfinite check early.  I expect the first few elements to be finite here; but I'm having to check the whole array feels wasteful.